geniushubclient/zone.py

```diff
@@ -33,15 +34,128 @@ def alphanum_key(k):
     return sorted(dict_list, key=alphanum_key)
 
 
+@dataclass
+class CurrentState:
+    """The current state of the zone.
+
+        Intended to be properties that can change in the lifecycle of the zone.
+    """
+
+    is_active: bool = False
+    is_requesting_heat: bool = False
+    mode: int = 0
+    mode_name: str = ""
+    temperature: float = 0.0
+    set_point: float = 0.0
+    is_occupied: bool = False
+
+
+@dataclass
+class Properties:
+    """The properties of the zone.
+
+        Intended to be the properties that are fixed for the lifecycle of the zone
+    """
+
+    type: int = 0
+    type_name: str = ""
+    has_room_sensor: bool = False
+
+
+@dataclass
+class DaySchedule:
+    """The day in a schedule. Contains a list of heating periods."""
+
+    index: int
```

Comment:
Done! Also made the name a property that is calculated from the index, rather than making the calling code do the calculation. Initialised the List too, so now the constructor only requires two parameters sending in.